index.js

```diff
@@ -14,7 +14,7 @@ function convertTypesToNamedTypes(TYPES) {
       formats[format.format] = format
     }
 
-    return Object.assign(type, { formats })
+    return Object.assign({}, type, { formats })
```

Comment:
Lets just merge this PR, there's nothing wrong with Object.assign, but just as a tip: could use object spread, it's a syntax supported in node.js >=8 and in all browsers except IE11. So `{...type, formats}`

Reply:
Ahh right, that is much nicer. Thanks! :)